lib/ice_cube/validations/monthly_by_set_pos.rb

```diff
@@ -0,0 +1,76 @@
+module IceCube
+
+  module Validations::MonthlyBySetPos
+    def by_set_pos(*by_set_pos)
+      by_set_pos.flatten!
+      by_set_pos.each do |set_pos|
+        unless (-366..366).include?(set_pos) && set_pos != 0
+          raise ArgumentError, "Expecting number in [-366, -1] or [1, 366], got #{set_pos} (#{by_set_pos})"
+        end
+      end
+
+      @by_set_pos = by_set_pos
+      replace_validations_for(:by_set_pos, [Validation.new(by_set_pos, self)])
+      self
+    end
+
+    class Validation
+
+      attr_reader :rule, :by_set_pos
+
+      def initialize(by_set_pos, rule)
+        @by_set_pos = by_set_pos
+        @rule = rule
+      end
+
+      def type
+        :day
+      end
+
+      def dst_adjust?
+        true
+      end
+
+      def validate(step_time, start_time)
+        start_of_month = step_time.beginning_of_month
```

Comment:
The `beginning_of_month` method is not available in vanilla Ruby. It's from active support (defined [here](https://github.com/rails/rails/blob/main/activesupport/lib/active_support/core_ext/date_and_time/calculations.rb#L125)). I tried testing this PR out in a non-Rails IRB session and I get an undefined method error here because active support is not available. This PR either needs to add a dependency on the active support gem or it needs re-implement `beginning_of_month` with only vanilla Ruby. I'm not a maintainer of this gem (just an interested party that would like to help get this functionality merged) but I suspect that adding a new dependency on active support may not be something they'd be willing to accept since this gem currently has no dependencies.

Reply:
Without using active support, I think this could be:

```ruby
start_of_month = IceCube::TimeUtil.build_in_zone([step_time.year, step_time.mo
nth, 1, 0, 0, 0], step_time)
```

Reply:
Inspiration for this approach was taken from the existing `beginning_of_date` method (defined [here](https://github.com/ice-cube-ruby/ice_cube/blob/master/lib/ice_cube/time_util.rb#L137)).

Reply:
Maybe it's even worth creating `beginning_of_xxx` and `end_of_xxx` helpers in `IceCube::TimeUtil` that are equivalent to their active support counterparts and then used them in the code here.

Reply:
Nice catch.  Pushed 621bf3db42b20b38563da1cb70ca202bc6ce6f4a to resolve this.